source/boot_security.c

```diff
@@ -11,64 +11,86 @@
 * Unauthorised copying of this file, via any medium is strictly prohibited.
 *
 ******************************************************************************/
-
-
- /******************************** VARIABLES ***********************************/
-
- //static const uint8_t Decryption_Key[DECRYPTION_KEY_LEN_IN_BYTES];
- // uint8_t* input = package->image;
- // uint8_t* output = input;
+ 
 
  
 /********************************* INCLUDES ***********************************/
 #include "boot_security.h"
+
 #include "aes.h"
+
 #include "hal_log.h"
 
- /******************************* MACRO DEFINITIONS ******************************/
-/*
-* Symmetric Key Length for Decryption
-* We use AES-256 for encryption/decryption
-*/
+
+/******************************* MACRO DEFINITIONS ******************************/
+
+    /*
+    * Symmetric Key Length for Decryption
+    * We use AES-256 for encryption/decryption
+    */
+
+
 #define DECRYPTION_KEY_LEN_IN_BITS 256
 #define DECRYPTION_KEY_LEN_IN_BYTES (DECRYPTION_KEY_LEN_IN_BITS / 8)
 
+/***************************** TYPE DEFINITIONS *******************************/
+
+/**************************** FUNCTION PROTOTYPES *****************************/
+    
+bool boot_authenticate_upgrade_package(boot_upgrade_package_t* package);
+
+void boot_decrypt_upgrade_package(boot_upgrade_package_t* package);
+
+/******************************** VARIABLES ***********************************/
+
 static const uint8_t Decryption_Key[DECRYPTION_KEY_LEN_IN_BYTES];
-/*******************************************************************************/
+
+/**************************** PRIVATE FUNCTIONS ******************************/
+
 
 /****************************  PUBLIC FUNCTIONS ******************************/
+
+
+    /**
+    * @brief boot_authenticate_upgrade_package function does something
+    * @param package[out] Argument 1 to get something out
+    * @retval true
+    */
+
```

Comment:
Please remove the above function header, the other group will be filling it. Thanks